test/integration/good/code-gen/cpp.expected

```diff
@@ -877,20 +901,43 @@ static constexpr std::array<const char*, 11> locations_array__ =
   " (in 'complex-tuples.stan', line 15, column 2 to column 74)",
   " (in 'complex-tuples.stan', line 3, column 4 to column 27)",
   " (in 'complex-tuples.stan', line 2, column 105 to line 4, column 3)"};
+template <typename T0__, typename T1__0__0__, typename T1__0__1__,
+          typename T1__1__,
+          stan::require_all_t<std::is_integral<T0__>,
+                              std::is_integral<T1__0__0__>,
+                              std::is_integral<T1__0__1__>,
+                              stan::is_std_vector<T1__1__>,
+                              stan::is_std_vector<stan::value_type_t<T1__1__>>,
+                              std::is_integral<stan::value_type_t<stan::value_type_t<T1__1__>>>>* = nullptr>
 std::tuple<int, std::vector<std::tuple<int, double>>>
-f(const int& x,
+f(const T0__& x,
   const std::vector<
           std::vector<
-            std::tuple<std::vector<std::tuple<int, int>>,
-              std::vector<std::vector<int>>>>>& x2, std::ostream* pstream__);
+            std::tuple<std::vector<std::tuple<T1__0__0__, T1__0__1__>>,
+              T1__1__>>>& x2, std::ostream* pstream__);
+/* tuple(int, array[] tuple(int, real)) f(int,
+                                         array[,] tuple(array[] tuple(int,
+  int),
+                                                        array[,] int))
+ */
```

Comment:
(optional) if we can figure out a way to not make this so funky that would be nice